### PR TITLE
Add support for Decimal Flat Vector

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

File: velox/vector/tests/VectorMakerStats.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include <optional>
 #include <unordered_set>
 
+#include "velox/type/ShortDecimal.h"
```
Comment (majetideepak) [inline velox/vector/tests/VectorMakerStats.h:22]: clean up.

File: velox/vector/tests/VectorMaker.h [hunk 2]
```diff
@@ -193,6 +193,19 @@ class VectorMaker {
     return flatVector;
   }
 
+  /// Create a DECIMAL FlatVector from native DECIMAL values, precision, scale.
+  ///
+  /// Elements are nullable.
+  ///
+  /// Examples:
+  ///   auto flatVector = decimalFlatVectorNullable({ShortDecimal(1),
```
Comment (mbasmanova) [inline velox/vector/tests/VectorMaker.h:201]: It is a bit verbose to wrap each value in a ShortDecimal. Is there a way to allow for shorter syntax?

```
decimalFlatVectorNullable({1, 2, std::nullopt, 5}, DECIMAL(10, 1))
```

File: velox/vector/tests/VectorMakerStats.h (hunk 1 above)
Comment (mbasmanova) [inline velox/vector/tests/VectorMakerStats.h]: Is this include needed?

File: velox/vector/tests/VectorMaker.h (hunk 2 above)
Comment (majetideepak) [reply velox/vector/tests/VectorMaker.h:201]: yeah, I felt the same. Will improve.

Comment (majetideepak) [reply velox/vector/tests/VectorMaker.h]: done.

File: velox/vector/tests/VectorMaker.h [hunk 3]
```diff
@@ -193,6 +193,22 @@ class VectorMaker {
     return flatVector;
   }
 
+  /// Create a DECIMAL FlatVector from native DECIMAL values, precision, scale.
+  ///
+  /// Elements are nullable.
+  ///
+  /// Examples:
+  ///  auto flatVector = decimalFlatVectorNullable({1, std::nullopt, 3}, 8, 1);
```
Comment (mbasmanova) [inline velox/vector/tests/VectorMaker.h:201]: I find 8, 1 not very readable. Would it make sense to pass DECIMAL(8, 1) instead?

File: velox/vector/tests/VectorMaker.h [hunk 4]
```diff
@@ -193,6 +193,22 @@ class VectorMaker {
...
+  FlatVectorPtr<ShortDecimal> decimalFlatVectorNullable(
```
Comment (mbasmanova) [inline velox/vector/tests/VectorMaker.h:202]: Would it be helpful to provide a non-nullable version as well to allow for shorter code when nulls are not used?

File: velox/vector/tests/VectorToStringTest.cpp [hunk 5]
```diff
@@ -99,4 +99,35 @@ TEST_F(VectorToStringTest, opaque) {
       "1: <opaque>\n"
       "2: <opaque>");
 }
+
+TEST_F(VectorToStringTest, decimals) {
+  std::vector<std::optional<int64_t>> shortValues = {
```
Comment (mbasmanova) [inline velox/vector/tests/VectorToStringTest.cpp:104]: We can remove this variable for readability.

File: velox/vector/tests/VectorToStringTest.cpp [hunk 6]
```diff
@@ -99,4 +99,35 @@ TEST_F(VectorToStringTest, opaque) {
...
+      "[FLAT SHORT_DECIMAL(10,3): 5 elements, 1 nulls]");
+  ASSERT_EQ(
+      shortDecimal->toString(0, 5),
+      "0: 1000.265\n"
+      "1: 35.610\n"
+      "2: -314.159\n"
+      "3: 0.007\n"
+      "4: null");
+
+  std::vector<std::optional<int128_t>> longValues = {
```
Comment (mbasmanova) [inline velox/vector/tests/VectorToStringTest.cpp:118]: same here

File: velox/vector/tests/VectorTestBase.h [hunk 7]
```diff
@@ -166,6 +166,22 @@ class VectorTestBase {
     return vectorMaker_.flatVector<T>(size, type);
   }
 
+  FlatVectorPtr<ShortDecimal> makeDecimalFlatVectorNullable(
```
Comment (mbasmanova) [inline velox/vector/tests/VectorTestBase.h:169]: inconsistent naming: makeNullableFlatVector vs. makeDecimalFlatVectorNullable

perhaps, rename to makeNullableFlatDecimalVector

File: velox/vector/tests/VectorMaker.h (hunk 2 above)
Comment (mbasmanova) [reply velox/vector/tests/VectorMaker.h]: Looks so much nicer. Thank you!

**Review by mbasmanova — APPROVED** [review]
@majetideepak This is nice. Thank you, Deepak, for iterating on a clean easy to read interface.

**Pull request closed without merging**